### PR TITLE
Add ability to use YellowFin optimizer

### PR DESCRIPTION
**Pull request by citerana**
This PR adds a Keras-wrapped TensorFlow optimizer called YellowFin. This optimizer automatically adjusts several hyperparameters while the model trains to converge towards some optimal confirguation. YellowFin takes an initial learning rate, but has optionally tuned momentum and learning_factor parameters.

In our experiments with the Planet Kaggle dataset, we found that the recommended `init_lr=1.0` and `momentum=0.0` from the paper performed poorly. It is unclear why YF behaved in such a way, but take the recommendations with a grain of salt and properly tune this optimizer to your models.

### CODE REVIEW

**Comment by lewfish** [comment]
If you fix merge conflicts, move `yellowfin.py` to `common/optimizers/yellowfin.py`, and add a comment in the file saying that we weren't able to get it to work and that it's "experimental" this should be good to go.
